Demo1.js

```diff
@@ -27,6 +27,15 @@ export default React.createClass({
     };
   },
 
+  handleTouchStart: function(key, pressLocation, event) {
+    this.handleMouseDown(key, pressLocation, event.touches[0]);
```

Comment:
Weeeeeell, basically I haven’t looked at destructuring arguments at all, and definitely not destructuring array arguments inside of objects.